### PR TITLE
fix: improve interactive session detection

### PR DESCRIPTION
**Pull request by DanStough**

### CODE REVIEW

File: scripts/common/common.sh [hunk 1]
```diff
@@ -1,6 +1,3 @@
-
-STORAGE_PROVISIONER=rook
-
```
Comment (DanStough) [inline scripts/common/common.sh:3]: Unused

File: scripts/common/preflights.sh [hunk 2]
```diff
@@ -365,14 +365,10 @@ function host_preflights() {
...
         # interactive terminal
-        if [ -t 0 ] ; then
+        if canPrompt; then      
             if ! "${DIR}"/bin/kurl host preflight "${MERGED_YAML_SPEC}" ${opts} </dev/tty ; then
-                printf "${RED}Host preflights have failures. Do you want to proceed anyway? ${NC} "
-                if ! confirmN "-t 30"; then
-                    report_install_fail "preflight"
-                    bail "Use the \"preflight-ignore\" flag to proceed."
-                fi
-                # TODO: report preflight fail
```
Comment (DanStough) [inline scripts/common/preflights.sh:375]: `kurl` already prompts the user here, although the default is backward from what we wanted in the script (Yes is default in the binary)

File: scripts/common/common.sh [hunk 3]
```diff
@@ -709,3 +706,12 @@ MACHINE_ID=
 function get_machine_id() {
     MACHINE_ID="$(${DIR}/bin/kurl host protectedid || true)"
 }
+
+function canPrompt() {
```
Comment (emosbaugh) [inline scripts/common/common.sh:710]: can you make this snake case?

File: scripts/common/preflights.sh [hunk 4]
```diff
@@ -365,14 +365,10 @@ function host_preflights() {
         # TODO: report preflight fail
     else
         # interactive terminal
-        if [ -t 0 ] ; then
+        if canPrompt; then
```
Comment (emosbaugh) [inline scripts/common/preflights.sh:368]: remove trailing whitespace

File: scripts/common/preflights.sh (hunk 2 above)
Comment (emosbaugh) [reply scripts/common/preflights.sh:375]: but what if there are failures?

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: scripts/common/common.sh (hunk 3 above)
Comment (DanStough) [reply scripts/common/common.sh]: Np. There are some other places we'll need to make these changes too like `prompts.sh`

File: scripts/common/preflights.sh (hunk 2 above)
Comment (DanStough) [reply scripts/common/preflights.sh]: Yes, I failed to consider that case. You'll get a double prompt if you hit No to exit on the warnings. I think the solution could be using different exits codes in `kurl`. I looked at this for a second and might need some help dealing with the errors package if we want to go that route. I reverted it for now.

File: scripts/common/common.sh [hunk 5]
```diff
@@ -1,6 +1,3 @@
-
-STORAGE_PROVISIONER=rook
```
Comment (emosbaugh) [inline scripts/common/common.sh:2]: Why is this going away?

File: scripts/common/common.sh (hunk 3 above)
Comment (emosbaugh) [reply scripts/common/common.sh]: Ya, I think for all new code we should stick to snake

File: scripts/common/common.sh (hunk 5 above)
Comment (DanStough) [reply scripts/common/common.sh:2]: `STORAGE_PROVISIONER` is unused anywhere in the code

**Review by emosbaugh — APPROVED** [review]

File: scripts/common/preflights.sh [hunk 6]
```diff
@@ -365,16 +365,25 @@ function host_preflights() {
...
     else
         # interactive terminal
-        if [ -t 0 ] ; then
-            if ! "${DIR}"/bin/kurl host preflight "${MERGED_YAML_SPEC}" ${opts} </dev/tty ; then
-                printf "${RED}Host preflights have failures. Do you want to proceed anyway? ${NC} "
-                if ! confirmY "-t 10"; then
+        if can_prompt; then
+            set +e
+            "${DIR}"/bin/kurl host preflight "${MERGED_YAML_SPEC}" </dev/tty
+            local kurlExitCode=$?
```
Comment (emosbaugh) [inline scripts/common/preflights.sh:371]: snake case?

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**